### ISSUE TITLE
Add built-in backup function

### ISSUE THREAD

**Issue by czlee** [czlee/tabbycat#25]
Backup function would be user-driven via web interface, possibly they could download a file or something.

**Comment by czlee** [comment]
I thought about this and did some more digging. If it's possible to invoke backups from within the app it's not obvious how to do so. The closest I can find is this app, which adds management commands for the purpose, writing backups to a file on the machine: https://django-dbbackup.readthedocs.io/en/stable/
and I'm not sure that Heroku plays nicely with user-generated files on the system, even if it weren't for [this StackOverflow question indicating that it doesn't work](http://stackoverflow.com/questions/24025150/running-django-dbbackup-on-heroku-yields-no-such-file-or-directory-error) (which isn't unexpected).

The better solution is probably to use Heroku's PGBackups like we do manually now. Heroku allows for scheduled backups, which are about as close as one can get to automatic: https://devcenter.heroku.com/articles/heroku-postgres-backups#scheduling-backups
but you need Heroku Toolbelt to do this; as far as I can tell you can't schedule backups from the Dashboard.

Bottom line is that I don't think it's really possible to add a built-in backup function. The old Tabbie1 used to generate an SQL dump for download, but I think the concept's fallen out of practice.

We could add a page to our documentation with instructions on how to do backups, or links to the relevant Heroku pages. The effect of this is that it's an "advanced use"—I don't really expect it to be accessible to people running small tournaments. But I think that's okay.
